.github/workflows/markdown.yml

```diff
@@ -0,0 +1,54 @@
+name: Markdown
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/markdown.yml'
+      - '.github/markdownlint.json'
+      - '.github/markdownlint.jsonc'
+      - '**/*.md'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  linter:
+    name: Lint Markdown Files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
```

Comment:
Just curious what this is good for. Is this something we should do for each workflow?

Reply:
You are right this needed to be there in all the workflows, This GitHub Action is a security tool designed to lock down GitHub Actions runners during a workflow execution. It helps prevent supply chain attacks by restricting outbound network traffic and tightening permissions dynamically. 

One of that action point I am planning to add https://github.com/ossf/scorecard to this repo. Which points out all these missing security checks.

Reply:
What's your recommendation: add the scorecard first? Or add the checks first?

Reply:
I recommend adding the checks first. But that would be a separate PR.

Reply:
Makes sense, thanks